pilot/pkg/model/telemetry_test.go

```diff
@@ -183,76 +226,159 @@ func TestAccessLogging(t *testing.T) {
 	tests := []struct {
 		name             string
 		cfgs             []config.Config
+		class            networking.ListenerClass
 		proxy            *Proxy
 		defaultProviders []string
 		want             []string
 	}{
 		{
 			"empty",
 			nil,
+			networking.ListenerClassSidecarOutbound,
 			sidecar,
 			nil,
 			nil,
 		},
 		{
 			"default provider only",
 			nil,
+			networking.ListenerClassSidecarOutbound,
 			sidecar,
 			[]string{"envoy"},
 			[]string{"envoy"},
 		},
 		{
 			"provider only",
 			[]config.Config{newTelemetry("istio-system", envoy)},
+			networking.ListenerClassSidecarOutbound,
+			sidecar,
+			nil,
+			[]string{"envoy"},
+		},
+		{
+			"client - gateway",
+			[]config.Config{newTelemetry("istio-system", client)},
+			networking.ListenerClassGateway,
+			sidecar,
+			nil,
+			[]string{"envoy"},
+		},
+		{
+			"client - outbound",
+			[]config.Config{newTelemetry("istio-system", client)},
+			networking.ListenerClassSidecarOutbound,
+			sidecar,
+			nil,
+			[]string{"envoy"},
+		},
+		{
+			"client - inbound",
+			[]config.Config{newTelemetry("istio-system", client)},
+			networking.ListenerClassSidecarInbound,
+			sidecar,
+			nil,
+			[]string{},
+		},
+		{
+			"server - gateway",
+			[]config.Config{newTelemetry("istio-system", server)},
+			networking.ListenerClassGateway,
+			sidecar,
+			nil,
+			[]string{},
+		},
+		{
+			"server - inbound",
+			[]config.Config{newTelemetry("istio-system", server)},
+			networking.ListenerClassSidecarInbound,
+			sidecar,
+			nil,
+			[]string{"envoy"},
+		},
+		{
+			"server - outbound",
+			[]config.Config{newTelemetry("istio-system", server)},
+			networking.ListenerClassSidecarOutbound,
+			sidecar,
+			nil,
+			[]string{},
+		},
+		{
+			"server and client - gateway",
+			[]config.Config{newTelemetry("istio-system", serverAndClient)},
+			networking.ListenerClassGateway,
+			sidecar,
+			nil,
+			[]string{"envoy"},
+		},
+		{
+			"server and client - inbound",
+			[]config.Config{newTelemetry("istio-system", serverAndClient)},
+			networking.ListenerClassSidecarInbound,
+			sidecar,
+			nil,
+			[]string{"envoy"},
+		},
+		{
+			"server and client - outbound",
+			[]config.Config{newTelemetry("istio-system", serverAndClient)},
+			networking.ListenerClassSidecarOutbound,
```

Comment:
can we add a test case(s) for something with multiple telemetry resources at different levels in the hierarchy (mesh wide, ns wide)?

Maybe something like:

```golang
[]config.Config{newTelemetry("istio-system", server), newTelemetry("foo", client)}
networkingListenerClassSidecar{Inbound|Outbound}
```

to validate proper overrides with differing modes?

Reply:
done, add tests with multiple telemetry resources.